esmvalcore/preprocessor/_multimodel.py

```diff
@@ -578,12 +601,12 @@ def _multicube_statistics(cubes, statistics, span, ignore_scalar_coords=False):
         )
 
     # Avoid modifying inputs
-    copied_cubes = [cube.copy() for cube in cubes]
+    cubes = [cube.copy() for cube in cubes]
```

Comment:
I don't understand the purpose of this - if you are dereferencing the initial `cubes` then you lost them, and only the garbage collector knows about them now

Reply:
The purpose of renaming `copied_cubes` to `cubes` is to make the code easier to understand. Using fewer variable names reduces the cognitive load: instead of remembering what `cubes`, `copied_cubes`, and `aligned_cubes` are, the reader now only needs to remember that `cubes` is the input data to the statistics computation.

Reply:
OK but then why copy them at all? Destroying the original `cubes` and copying them to a new address is a bit of an excercise in futility IMO but maybe am missing something - at any rate, don't let this hold the merge :+1:

Reply:
indeed - it's just a pointer in a dict :grin: